lfric_macros/apply_macros.py

```diff
@@ -788,6 +815,112 @@ def write_new_macro(self, meta_dir, full_command):
 
         os.rename(temppath, filepath)
 
+    def check_missing_macros(self, meta_dir, meta_imports):
+        """
+        Check through macros of imported metadata sections, returning list of any that
+        aren't in the current section (identified by the after tag)
+        Inputs:
+            - meta_dir, the current metadata section
+            - meta_imports, a list of imported metadata sections
+        Returns:
+            - list of macros that need adding to the current section
+        """
+
+        after_tags = [m["after_tag"] for m in self.parsed_macros[meta_dir]]
+
+        missing_macros = []
+        for section in meta_imports:
+            section = self.get_full_import_path(section)
+            section_missing = []
+            section_macros = []
+            for macro in self.parsed_macros[section]:
+                after_tag = macro["after_tag"]
+                # Ignore the macro being upgraded - this is expected to be missing
+                if after_tag == self.tag:
+                    continue
+                section_macros.append(after_tag)
+                if after_tag not in after_tags:
+                    section_missing.append(after_tag)
```

Comment:
```suggestion
                if after_tag == self.tag:
                    num_section_macros -= 1
                elif after_tag not in after_tags:
                    section_missing.append(after_tag)
```

If you go with the above suggestion then this section simplifies